x/nft/helper.go

```diff
@@ -12,22 +10,11 @@ const (
 	maxIDLength = 256
 )
 
-var (
-	isValidAction = regexp.MustCompile(`^[A-Za-z]{4,32}$`).MatchString
-)
-
-type Validation struct {
-}
-
-func (*Validation) IsValidAction(action string) bool {
-	return isValidAction(action)
```

Comment:
Why a regular expression was used? We know that there are only 3 valid actions that this implementation understands so we can just compare the strings. Should it be allowed to pass an action that is unknown to the implementation?

Reply:
yes, the reason we are accepting a string there and not a protobuf message is that we want to be able to extend on nft functionality from outside weave context when creating new apps.

For that same reason there was no explicit validation of existing actions using generated protobuf name map.

Reply:
So we do need a regex to make sure users implement these constants according to rules.
But, I think we also need, at this point, to be able to supply a list of all actions for validation.

You can take a look at ```func RegisterNft()``` in `cmd/bnsd/app.go` for inspiration, although we can instead register an array/list as we don't have `name -> bucket` pairs there. Or you can come up with something better if you have a good idea.

Reply:
If I understand correctly, it is not possible to fully validate the action string. We can check if it follows certain rules (regexp check), but we can never ensure that given action is supported, because `x/nft` user may use custom actions.

If above is true, I will bring back the regexp check. I understand the use case, but that means that action strings are not verified during the compilation phase and typo can still slip through. The only protection from this that I can think of is to have a global register that an action string must be registered in before it can be used.

Reply:
We need to prevent users from granting permissions on actions that are non-existent, this is a hole that would allow to abuse our storage system otherwise. So we'd need to validate the concrete actions too, that can be registered on the app level similar to nft types and their corresponding buckets. @alpe does that make sense?